src/Particular.LicensingComponent/ThroughputCollector.cs

```diff
@@ -89,19 +89,33 @@ public async Task<List<EndpointThroughputSummary>> GetThroughputSummary(Cancella
         return endpointSummaries;
     }
 
-    public async Task<ReportGenerationState> GetReportGenerationState(CancellationToken cancellationToken)
+    public async Task<ReportGenerationState> GetReportGenerationState(CancellationToken cancellationToken) =>
+        throughputQuery == null ?
+            await GetReportGenerationStateForNonBroker(cancellationToken) :
+            await GetReportGenerationStateForBroker(cancellationToken);
+
+    async Task<ReportGenerationState> GetReportGenerationStateForBroker(CancellationToken cancellationToken)
     {
-        var reportGenerationState = new ReportGenerationState(transport)
+        var reportCanBeGenerated = await dataStore.IsThereThroughputForLastXDaysForSource(30, ThroughputSource.Broker, false, cancellationToken);
+        var reason = reportCanBeGenerated ? string.Empty : $"Need at least one day of usage data in the last 30 days from ${throughputQuery!.MessageTransport} broker.";
```

Comment:
Still not exactly like what SP does https://github.com/Particular/ServicePulse/blob/ca4c4c8700a53051052151a13a0e86ef21189d83/src/Frontend/src/stores/ThroughputStore.ts#L54, but I guess it is better

Reply:
for Sql Server actually displays "SqlTransport" 😞

Reply:
I think I would make this generic and just say "Need at least one day of usage data in the last 30 days from the configured broker."